techmatters/hrm
# Expose permissions API

This PR is a proposal to expose the permissions rules for our helplines from HRM backend app.
Reopen of https://github.com/techmatters/hrm/pull/179

## Description
This PR
- Moves most files related to permissions framework to TS.
- Factors out the rules map (`{ [key: string]: RulesFile }`) to consume the rules definitions files while being imported from a single place.
- Adds a `/permissions` endpoint that will retrieve the permissions rules for the given account (if they are defined).

The idea of this PR is to expose a way to retrieve the permissions rules based on the configuration given to each helpline. For this purpose, I'm using the `accountSid` appended to each request to find the correct set of permissions (or throw if there is none in env vars).
Another PR should change Flex Plugin to consume from this endpoint on initialization, and store the permissions in local memory (probably in Rudex state, under config reducer). This way we'll get rid of the duplicate permissions files, and they'll only liver in here.

### Checklist

### Verification steps

- Check that all the backend side permissions are still in place.
- Start local server with a valid `PERMISSIONS_<accountSid>=<some permissions file key>`.
- Perform a GET on `'http://localhost:3000/v0/accounts/<accountSid>/permissions'` with header `-H "Authorization: Bearer <valid token>"` to retrieve the rules corresponding to `<some permissions file key>`.
- Change `PERMISSIONS_<accountSid>` value to some invalid key and hit again to reach error.
